docs/import/color_function_util.js

```diff
@@ -1,8 +1,162 @@
-import {ColorStyle} from '../firebase_layers.js';
+import {colorMap, ColorStyle} from '../firebase_layers.js';
+import {getCurrentLayers, getRowIndex, ILLEGAL_STATE_ERR, setStatus, updateLayersInFirestore} from './add_disaster_util.js';
 
-import {ILLEGAL_STATE_ERR, setStatus} from './add_disaster_util.js';
+export {populateColorFunctions, withColor};
 
-export {withColor};
+// At any given point in time, the color function div is displaying info
+// about a single asset. We use this global cell to keep trick of which
+// color function info is currently displayed.
+let globalTd;
+
+/**
+ * Fills out the #single #continuous and #discrete divs with the relevant
+ * DOM elements with attached on change handlers.
+ */
+function populateColorFunctions() {
+  const colorFunctionDiv = $('#color-fxn-editor');
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.SINGLE));
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.DISCRETE));
+  colorFunctionDiv.prepend(createRadioFor(ColorStyle.CONTINUOUS));
+
+  const singleColorPicker = createColorPicker('single-color-picker');
+  singleColorPicker.on('change', () => setColor(singleColorPicker));
+  $('#single').append(
+      createLabelFor(singleColorPicker, 'color: '), singleColorPicker);
+
+  const continuousColorPicker = createColorPicker('continuous-color-picker');
+  continuousColorPicker.on('change', () => setColor(continuousColorPicker));
+  const continuousPropertyPicker =
+      $(document.createElement('select'))
+          .prop('id', 'continuous-property-picker');
+  continuousPropertyPicker.on(
+      'change', () => setProperty(continuousPropertyPicker));
+  $('#continuous')
+      .append(
+          createLabelFor(continuousColorPicker, 'base color: '),
+          continuousColorPicker, $(document.createElement('br')),
+          createLabelFor(continuousPropertyPicker, 'property: '),
+          continuousPropertyPicker);
+
+  const discretePropertyPicker = $(document.createElement('select'))
+                                     .prop('id', 'discrete-property-picker');
+  discretePropertyPicker.on('change', () => {
+    setProperty(discretePropertyPicker);
+    populateDiscreteColorPickers();
+  });
+  const discreteColorPickers =
+      $(document.createElement('ul')).prop('id', 'discrete-color-pickers');
+  $('#discrete')
+      .append(
+          createLabelFor(discretePropertyPicker, 'property: '),
+          discretePropertyPicker, discreteColorPickers);
+}
+
+/**
+ * Updates the 'field' property which is shared by the continuous and discrete
+ * color schemas.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function setProperty(picker) {
+  getColorFunction()['field'] = picker.val();
+  updateLayersInFirestore();
+}
+
+/**
+ * Updates an individual color choice for a single value in the discrete schema.
+ * @param {JQuery<HTMLElement>} picker
+ */
+function setDiscreteColor(picker) {
+  const colorFunction = getColorFunction();
+  const propertyValue = picker.data('value');
```

Comment:
nit: pull this 'value' and 'value' from the picker creation out to a common constant?